src/main/java/reactor/netty/transport/ProxyProvider.java

```diff
@@ -293,6 +303,12 @@ public final Builder address(Supplier<? extends InetSocketAddress> addressSuppli
 		@Override
 		public final Builder nonProxyHosts(String nonProxyHostsPattern) {
 			this.nonProxyHosts = nonProxyHostsPattern;
+			return nonProxyHostsPattern == null ? this : nonProxyHostsPredicate(new RegexShouldProxyPredicate(nonProxyHostsPattern));
```

Comment:
Let's add here `StringUtil.isNullOrEmpty(nonProxyHostsPattern) ? this : nonProxyHostsPredicate(new RegexShouldProxyPredicate(nonProxyHostsPattern));`